### PR TITLE
Update table using crud

### PR DESCRIPTION
**Pull request by Ngufor-emmanuella**
**In this section of my work i did the following changes using CRUD method**

1. Added a column species of type string to the animals' table and modified the schema.sql file.

2. I inserted a new set of data into the table and added those insertions into the data.sql file.

3. Used database transactions to modify and delete data from SQL.

4. Prepare complex queries that answered analytical questions.

The following are the screenshots with the results of the queries and with the results of your play with _transactions:
snapshots for unspecified
Screenshot (70)
snapshots for animals species column
_
Screenshot (75)
transaction for updating weight of animals
Screenshot (77)
transactions for questions
Screenshot (78)
Screenshot (79)

_changes implemented suggested by the reviewer
_
Screenshot (80)
Screenshot (81)

### CODE REVIEW

File: schema.sql [hunk 1]
```diff
@@ -7,4 +7,8 @@ CREATE TABLE animals (
     escape_attempts INT,
     neutered BOOLEAN,
     weight_kg DECIMAL
-);
\ No newline at end of file
+);
+
+-- add a column to the table
+ALTER TABLE animals
+ADD COLUMN specie VARCHAR(100);
```
Comment (sja-thedude) [inline schema.sql]: - [ ] Kindly fix the typo it should be species.

File: data.sql [hunk 2]
```diff
@@ -4,3 +4,12 @@ INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg)
 INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg) VALUES (' Gabumon', date '2018-11-15', 2, true, 8);
 INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg) VALUES ('Pikachu', date '2021-01-07', 1, false, 15.04);
 INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg) VALUES ('Devimon', date '2017-05-12', 5, true, 11);
+
+-- adding new data
+INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg) VALUES ('Charmander', date '2020-02-08', 0, false, 11);
+INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg) VALUES ('Plantmon', date '2021-11-15', 2, true, 5.7);
+INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg) VALUES ('Squirtle', date '1993-04-02', 3, false, 12.13);
+INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg) VALUES ('Angemon', date '2005-06-12', 1, true, 45);
```
Comment (sja-thedude) [inline data.sql]: - [ ] Kindly make sure that weight_kg values for Charmander, Plantmon, Squirtle, and Angemon are in negative. Like -11.0, not 11.

File: queries.sql [hunk 3]
```diff
@@ -7,4 +7,40 @@ SELECT date_of_birth FROM animals WHERE name='Agumon' OR name='Pikachu';
...
+COMMIT;
+
+-- WRITE COMMITPOINTQUERIES FOR QUESTIONS
+
+SELECT COUNT(name) AS total_animals FROM animals; 
+SELECT name, SUM(escape_attempts) FROM animals GROUP BY name having MIN(escape_attempts) < 1;
+SELECT AVG(weight_kg) FROM animals;
+SELECT neutered, MAX(escape_attempts) FROM animals GROUP BY neutered;
+SELECT species, MIN(weight_kg), MAX(weight_kg) FROM animals GROUP BY species;
+SELECT species, AVG(escape_attempts) FROM animals WHERE date_of_birth BETWEEN '1990-01-01' AND '2000-01-01' GROUP BY species;
```
Comment (sja-thedude) [inline queries.sql]: - [ ] Kindly make sure the end date is 2000-12-31 as its the entire 2000 year.

File: queries.sql [hunk 4]
```diff
@@ -7,4 +7,40 @@ SELECT date_of_birth FROM animals WHERE name='Agumon' OR name='Pikachu';
...
+
+-- update table in a transaction
+BEGIN;
+UPDATE animals SET species='digimon' WHERE name LIKE '%mon';
+UPDATE animals SET species = 'pokemon' WHERE species IS NULL;
+COMMIT;
+
+-- delete transaction in the table and roll back;
+DELETE FROM animals;
+ROLLBACK;
```
Comment (sja-thedude) [inline queries.sql]: - [ ] Kindly add `BEGIN` to begin this transaction also kindly add  `SELECT COUNT(*) FROM ANIMALS;` before and after ROLLBACK to verify if the change was made.

File: queries.sql [hunk 5]
```diff
@@ -7,4 +7,40 @@ SELECT date_of_birth FROM animals WHERE name='Agumon' OR name='Pikachu';
...
+
+-- DELETE COLUMN
+ALTER TABLE animals DELETE COLUMN species;
+
+-- update table in a transaction
+BEGIN;
+UPDATE animals SET species='digimon' WHERE name LIKE '%mon';
+UPDATE animals SET species = 'pokemon' WHERE species IS NULL;
+COMMIT;
```
Comment (sja-thedude) [inline queries.sql]: - [ ] Kindly add :
```sql
 SELECT species from animals; 
```
before COMMIT to verify that change was made and after COMMIT to verify that change persists after commit

File: queries.sql [hunk 6]
```diff
@@ -7,4 +7,40 @@ SELECT date_of_birth FROM animals WHERE name='Agumon' OR name='Pikachu';
...
 SELECT * FROM animals WHERE name <> ' Gabumon';
-SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
\ No newline at end of file
+SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
+
+-- update tables with transactions
+BEGIN TRANSACTION;
+UPDATE animals
+SET species = 'unspecified';
+ROLLBACK;
```
Comment (sja-thedude) [inline queries.sql]: - [ ] Kindly make sure to add query to verify the change was made before ROLLBACK and after ROLLBACK to verify if the change was undone.

```sql
   SELECT species from animals;
```

File: queries.sql [hunk 7]
```diff
@@ -7,4 +7,40 @@ SELECT date_of_birth FROM animals WHERE name='Agumon' OR name='Pikachu';
...
+
+-- DELETE COLUMN
+ALTER TABLE animals DELETE COLUMN species;
```
Comment (sja-thedude) [inline queries.sql]: - [ ] Kindly remove this it is already added in schema.sql and doesnt need here

**Review by sja-thedude — CHANGES_REQUESTED** [review]
## Required Changes ♻️

Hi @Ngufor-emmanuella, This is @sja-thedude your code reviewer.

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

768070f3-bc9a-4aba-b8bc-25793e00ea1e_text

## Highlights
✔️ Professional Readme
✔️ Correct Gitflow

## Required 
- [ ] Kindly remove the broken microverse image from readme.
- [ ] _Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me [@sja-thedude] in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: queries.sql (hunk 7 above)
Comment (Ngufor-emmanuella) [reply queries.sql]: hi, @sja-thedude thank you for the corrections. I already implemented the changes

File: data.sql (hunk 2 above)
Comment (rloterh) [reply data.sql]: well resolved! ✔️

File: queries.sql (hunk 4 above)
Comment (rloterh) [reply queries.sql]: well resolved! ✔️

File: queries.sql (hunk 3 above)
Comment (rloterh) [reply queries.sql]: well resolved! ✔️

File: queries.sql (hunk 6 above)
Comment (rloterh) [reply queries.sql]: well resolved! ✔️

File: queries.sql (hunk 5 above)
Comment (rloterh) [reply queries.sql]: well resolved! ✔️

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Emma,  

Your project is complete! 🏁 There is nothing else to say other than... it's time to merge it 🍾Congratulations!  🎉💯🌟 

## Highlights 👍
- Queries effectively added!

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

_Kindly consider my inline comments marked **[OPTIONAL]** under this review_

**You can also consider:**
- MIT license link in your ReadMe is broken, kindly endeavor to fix it, and remove the guide note below it.

Cheers..Happy coding! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Ngufor-emmanuella** [comment]
thank you

**Pull request merged**